src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -38,6 +38,10 @@ message ResourceCommand {
     // Clients must return any value provided by the server when invoking
     // the command.
     optional google.protobuf.Value parameter = 4;
+    // A flag that indicates whether the command is highlighted in the UI.
+    bool is_highlighted = 5;
+    // Optional icon content. SVG path is displayed in the UI.
+    optional string icon_content = 6;
```

Comment:
Defining the icon as SVG means we cannot easily show icons in all clients.

For example, if we want to add a tool window to VS to support interaction with an AppHost (essentially, creating a new kind of client for the resource service) then I don't think we'll be able to display these icons. All solutions I've seen for this involve bringing in other heavy libraries (like SkiaSharp or Xamarin.Forms) which we won't be able to do.

Can we use a richer type here that provides more opportunity for clients to understand the semantics of the icon and provide alternative images?

If we used the name of the FluentUI icon's class, we could instantiate it using reflection to get SVG content. That would have the benefit of modernising automatically when the design language changes, which happens every few years. We don't want the UI to look dated.

Passing an icon name also avoids my sibling comment about implicit size information.

Reply:
We could have the fluentui icon name, but those other clients couldn't use it either. All they could do is look at the icon name and substitute in their own icon.

However, if they want to add icons then nothing is stopping them from ignoring this icon content and basing the icon on the command type.

e.g. command type = "Start", UI displays its play icon. command type = "Stop", UI displays its stop icon.

Reply:
I prefer the icon name to sending svg directlY. I think orchard does the same thing